### PR TITLE
added "all" function - to pass in object and output filter result ("and")

### PR DESCRIPTION
**Pull request by tangallan**
- Updated package.json for test, adding --testEnvironment
- Created new function "compareAll" to map object to odata filter result
- Created "compareAll" function that reads key and values and join odata expression for "and"

### CODE REVIEW

**Comment by bodia-uz** [comment]
Hi @tangallan. Thanks for your PR.
LGTM.
One question to discuss. Mb we can choose some other name for the public method?
While most of the current methods are the same as OData operators, `filtarBuilder.all` could be not clear enough. Mb we could use `filtarBuilder.compareAll`, or any other suggestions?

File: src/lib/comparison.js [hunk 1]
```diff
@@ -67,6 +67,20 @@ function compareIn(field, values, normaliseValues) {
...
 }
 
+function compareAll(objectValue, normaliseValues) {
+  const keys = Object.keys(objectValue);
+  return joinRules(keys.map(k => {
+    const ov = objectValue[k];
+
+    if(Array.isArray(ov)) {
+      return `(${compareIn(k, ov, normaliseValues)})`;
+    } else {
```
Comment (bodia-uz) [inline src/lib/comparison.js]: What do you think about how to handle undefined values?
```
{
  name: 'john doe',
  age: undefined
}
```
It relevant to how should we handle `filterBuilder.eq('age', undefined)`(#18)

File: src/lib/comparison.js [hunk 2]
```diff
@@ -67,6 +67,20 @@ function compareIn(field, values, normaliseValues) {
...
+      return compare(k, 'eq', ov, normaliseValues);
```
Comment (bodia-uz) [inline src/lib/comparison.js]: as variant
```suggestion
      return eq(k, ov, normaliseValues);
```

File: src/lib/comparison.js [hunk 3]
```diff
@@ -67,6 +67,20 @@ function compareIn(field, values, normaliseValues) {
   return joinRules(compareMap(field, 'eq', values, normaliseValues), 'or');
 }
 
+function compareAll(objectValue, normaliseValues) {
+  const keys = Object.keys(objectValue);
+  return joinRules(keys.map(k => {
+    const ov = objectValue[k];
```
Comment (bodia-uz) [inline src/lib/comparison.js]: mb use `field` and `value` names? To make it similar as in other "operators"
```suggestion
    const value = objectValues[field];
```

**Comment by tangallan** [comment]
> Hi @tangallan. Thanks for your PR.
> LGTM.
> One question to discuss. Mb we can choose some other name for the public method?
> While most of the current methods are the same as OData operators, `filtarBuilder.all` could be not clear enough. Mb we could use `filtarBuilder.compareAll`, or any other suggestions?

@bodia-uz sorry for the late reply, for some reason I am not getting notifications on messages, but yes we can definitely rename the function, `compareAll` works for me actually - all might have been to vague, thanks I will fix this.

File: src/lib/comparison.js (hunk 3 above)
Comment (tangallan) [reply src/lib/comparison.js]: updated :)

File: src/lib/comparison.js (hunk 1 above)
Comment (tangallan) [reply src/lib/comparison.js]: @bodia-uz  - I filter out the undefined values (silently skipped the properties)

**Comment by bodia-uz** [comment]
Thanks. Merged.
Not sure when I will have ability to release it. Mb on next week.

**Comment by bodia-uz** [comment]
released in [v1.0.0](https://github.com/bodia-uz/odata-filter-builder/releases/tag/v1.0.0)

**Pull request merged**